test/ng/directive/ngModelSpec.js

```diff
@@ -1784,6 +1807,63 @@ describe('ngModelOptions attributes', function() {
   }));
 
 
+  it('should inherit options from ngModelOptions directives declared on ancestor elements', function() {
+    var container = $compile('<div ng-model-options="{ allowInvalid: true }">' +
+               '<form ng-model-options="{ updateOn: \'blur\' }">' +
+                 '<input ng-model-options="{ updateOn: \'default\' }">' +
```

Comment:
The idea is that we have three levels of options: container, form and input.
We want to check that the `allowInvalid` option is getting inherited all the way down. But I agree that we might not be creating a new `ModelOptions` object if this `<input>` does not contain an `ng-model` directive. So we should indeed add that.